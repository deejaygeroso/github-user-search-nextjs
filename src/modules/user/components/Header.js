import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import Avatar from '@material-ui/core/Avatar';
import red from '@material-ui/core/colors/red';
import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
import Button from '@material-ui/core/Button';
import Router from 'next/router'

const styles = {
  card: {
    display: 'flex',
    flexDirection: 'row',
    borderRadius: 0
  },
  avatar: {
    backgroundColor: red[500],
  },
  cardHeader: {
    paddingTop: 10,
    paddingBottom: 10,
    paddingLeft: 0
  },
  iconHover: {
    fontSize: 50,
    margin: 0,
  },
  button: {
    padding: 0,
    margin: 0,
  }, 
};

class Header  extends React.Component {
  state = { expanded: false };

  goBack = () => {
      Router.back()
  };

  render() {
    const { classes, user } = this.props;
    return (
      <Card className={classes.card}>
        <Button onClick={this.goBack} className={classes.button}>
          <KeyboardArrowLeft className={classes.iconHover} />
        </Button>
        <CardHeader
          avatar={
            <Avatar src={user.avatar_url} aria-label="Recipe" className={classes.avatar} />
          }
          title={user.login}
          subheader={user.html_url}
          className={classes.cardHeader}
        />
      </Card>
    );
  }
}

Header .propTypes = {
  user : PropTypes.object,
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Header );
