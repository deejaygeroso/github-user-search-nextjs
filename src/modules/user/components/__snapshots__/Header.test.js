import React from 'react';
import Header from '../Header';
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const user = {}
  const tree = renderer
    .create(<Header user={user} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});