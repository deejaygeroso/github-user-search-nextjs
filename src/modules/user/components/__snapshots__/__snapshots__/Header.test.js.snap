// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root-7 MuiPaper-elevation1-10 MuiPaper-rounded-8 MuiCard-root-6 Header-card-1"
>
  <button
    className="MuiButtonBase-root-60 MuiButton-root-34 MuiButton-text-36 MuiButton-flat-39 Header-button-5"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-35"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-63 Header-iconHover-4"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-81"
    />
  </button>
  <div
    className="MuiCardHeader-root-72 Header-cardHeader-3"
  >
    <div
      className="MuiCardHeader-avatar-73"
    >
      <div
        aria-label="Recipe"
        className="MuiAvatar-root-78 MuiAvatar-colorDefault-79 Header-avatar-2"
      />
    </div>
    <div
      className="MuiCardHeader-content-75"
    />
  </div>
</div>
`;
